File: app/emit_box.c
#include "app.h"

void
emit_box(void)
{
	double x1, x2, y1, y2;

	x1 = DRAW_LEFT_MARGIN;
	x2 = DRAW_LEFT_MARGIN + DRAW_WIDTH;

	y1 = 0.0;
	y2 = DRAW_HEIGHT;

	// left

	emit_push(DRAW_STROKE);
	emit_push(x1);
	emit_push(y1);
	emit_push(x1);
	emit_push(y2);
	emit_push(VERY_THIN_STROKE);

	// right

	emit_push(DRAW_STROKE);
	emit_push(x2);
	emit_push(y1);
	emit_push(x2);
	emit_push(y2);
	emit_push(VERY_THIN_STROKE);

	// top

	emit_push(DRAW_STROKE);
	emit_push(x1);
	emit_push(y1);
	emit_push(x2);
	emit_push(y1);
	emit_push(VERY_THIN_STROKE);

	// bottom

	emit_push(DRAW_STROKE);
	emit_push(x1);
	emit_push(y2);
	emit_push(x2);
	emit_push(y2);
	emit_push(VERY_THIN_STROKE);
}

File: js-experimental/emit_box.js
function
emit_box()
{
	var x1 = 0;
	var x2 = DRAW_WIDTH;

	var y1 = 0;
	var y2 = DRAW_HEIGHT;

	draw_line(x1, y1, x2, y1, 0.5); // top line
	draw_line(x1, y2, x2, y2, 0.5); // bottom line

	draw_line(x1, y1, x1, y2, 0.5); // left line
	draw_line(x2, y1, x2, y2, 0.5); // right line
}
